agent/engine/task_manager.go

```diff
@@ -606,6 +611,10 @@ func (mtask *managedTask) cleanupTask(taskStoppedDuration time.Duration) {
 	mtask.engine.sweepTask(mtask.Task)
 
 	// TODO: Remove platform task resources
+	err := mtask.CleanupPlatformResources()
```

Comment:
Is the todo here still needed? Also will this impact other new task if we don't remove this right after the task is stopped, since the cleanup happens after `TASK_CLEANUP_DURATION`?